app/controllers/application_controller.rb

```diff
@@ -325,19 +331,23 @@ def clear_notifications
     end
   end
 
-  def set_locale
-    if !current_user
+  def with_resolved_locale(check_current_user: true)
+    if check_current_user && current_user
```

Comment:
Do we need to check if `SiteSetting.allow_user_locale` is `true` here? I see it being set to `true` in the tests but it doesn't seem to be used in the controller.

Reply:
We have to set it `true` in the tests because `set_locale_from_accept_language_header` [has a validation for it](https://github.com/discourse/discourse/blob/db4e310376488e87eb6968df5bf7fa17099297ab/lib/validators/allow_user_locale_enabled_validator.rb#L9-L14).

`SiteSetting.allow_user_locale` is not used in the controller directly, but it's checked in [`User#effective_locale`](https://github.com/discourse/discourse/blob/db4e310376488e87eb6968df5bf7fa17099297ab/app/models/user.rb#L330-L334)

Reply:
Got it :) thanks for clarifying.